projects/admin-portal/src/app/components/request/request.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Request } from 'projects/shared/src/lib/models/request';
+import { FormControl } from '@angular/forms';
+import { NewCommentModel } from 'projects/shared/src/lib/models/new-comment-model';
+import { RequestStatus } from 'projects/shared/src/lib/models/request-status';
+import { HttpClient } from '@angular/common/http';
+import { Comment } from 'projects/shared/src/lib/models/comment';
+
+@Component({
+  selector: 'app-request',
+  templateUrl: './request.component.html',
+  styleUrls: ['./request.component.scss']
+})
+export class RequestComponent implements OnInit {
+
+  private roomId: number;
+  private requestId: number;
+
+  constructor(private route: ActivatedRoute, private router: Router, private http: HttpClient) {
+    this.route.data.subscribe(data => {
+      this.requestInfo = data.requestInfo;
+      this.requestInfo.comments.reverse();
+    });
+
+    this.roomNumber = this.route.snapshot.queryParams.roomNumber;
+    this.floorNumber = this.route.snapshot.queryParams.floorNumber;
+    this.roomId = this.route.snapshot.queryParams.roomId;
+    this.requestId = this.route.snapshot.queryParams.requestId;
+  }
+
+  public requestInfo: Request;
+  public newComment: FormControl;
+  public floorNumber: number;
+  public roomNumber: string;
+
+  ngOnInit() {
+    this.newComment = new FormControl();
+  }
+
+  public backToRoom() {
+    this.router.navigateByUrl(`rooms?floorNumber=${this.floorNumber}&roomId=${this.roomId}&roomNumber=${this.roomNumber}`);
+  }
+
+  public backToFloor() {
+    this.router.navigateByUrl(`floors/${this.floorNumber}`);
+  }
+
+  public sendAcceptComment() {
+    const newComment: NewCommentModel = new NewCommentModel();
```

Comment:
You don't need to declare the type if you are creating a model here. And why don't you pass following properties as a parameter t the constructor?